src/fm/cp_fm_elpa.F

```diff
@@ -75,33 +81,50 @@ MODULE cp_fm_elpa
 
    CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'cp_fm_elpa'
 
+! The size of an array with the attribute parameter has to be explicitly passed the array size
+! This ugly constructions allows us to deal with different array sizes to partially missing constants
 #if defined(__ELPA)
-   INTEGER, DIMENSION(21), PARAMETER :: elpa_kernel_ids = [ &
-                                        ELPA_2STAGE_REAL_INVALID, & ! auto
-                                        ELPA_2STAGE_REAL_GENERIC, &
-                                        ELPA_2STAGE_REAL_GENERIC_SIMPLE, &
-                                        ELPA_2STAGE_REAL_BGP, &
-                                        ELPA_2STAGE_REAL_BGQ, &
-                                        ELPA_2STAGE_REAL_SSE_ASSEMBLY, &
-                                        ELPA_2STAGE_REAL_SSE_BLOCK2, &
-                                        ELPA_2STAGE_REAL_SSE_BLOCK4, &
-                                        ELPA_2STAGE_REAL_SSE_BLOCK6, &
-                                        ELPA_2STAGE_REAL_AVX_BLOCK2, &
-                                        ELPA_2STAGE_REAL_AVX_BLOCK4, &
-                                        ELPA_2STAGE_REAL_AVX_BLOCK6, &
-                                        ELPA_2STAGE_REAL_AVX2_BLOCK2, &
-                                        ELPA_2STAGE_REAL_AVX2_BLOCK4, &
-                                        ELPA_2STAGE_REAL_AVX2_BLOCK6, &
-                                        ELPA_2STAGE_REAL_AVX512_BLOCK2, &
-                                        ELPA_2STAGE_REAL_AVX512_BLOCK4, &
-                                        ELPA_2STAGE_REAL_AVX512_BLOCK6, &
-                                        ELPA_2STAGE_REAL_NVIDIA_GPU, &
-                                        ELPA_2STAGE_REAL_AMD_GPU, &
-                                        ELPA_2STAGE_REAL_INTEL_GPU_SYCL]
+   INTEGER, DIMENSION(18 &
+#if defined (__ELPA_NVIDIA_GPU)
+                      + 1 &
+#endif
+#if defined (__ELPA_AMD_GPU)
+                      + 1 &
+#endif
+#if defined (__ELPA_INTEL_GPU)
+                      + 1 &
+#endif
```

Comment:
Why is this needed? AFAIK, those constants are always available - even when the GPU kernels are not.

Reply:
The `ELPA_2STAGE_REAL_INTEL_GPU_SYCL` kernel was added only recently. So, you are probably using an older version of ELPA. Given that ELPA had a number of sever issue in the past, I'm not sure we should actually support backwards compatibility.

Reply:
Then, we can actually remove these kind of checks.

Reply:
I removed the respective commit.